mne/time_frequency/_stockwell.py

```diff
@@ -32,16 +31,16 @@ def _check_input_st(x_in, n_fft, verbose=None):
     if n_times < n_fft:
         msg = ('The input signal is shorter ({0}) than "n_fft" ({1}). '
                'Applying zero padding.').format(x_in.shape[-1], n_fft)
-        logger.warn(msg)
+        logger.warning(msg)
         zero_pad = n_fft - n_times
-        pad_array = np.zeros(list(x_in.shape[:-1]) + [zero_pad], dtype=x_in.dtype)
+        pad_array = np.zeros(list(x_in.shape[:-1]) + [zero_pad], x_in.dtype)
```

Comment:
but just for the record, `git blame` says this one is @agramfort's fault originally :)